src/test_l3_dsDisplay.c

```diff
@@ -0,0 +1,330 @@
+    {"dsVIDEOPORT_TYPE_RF", (int32_t)dsVIDEOPORT_TYPE_RF},
+    {"dsVIDEOPORT_TYPE_BB", (int32_t)dsVIDEOPORT_TYPE_BB},
+    {"dsVIDEOPORT_TYPE_SVIDEO", (int32_t)dsVIDEOPORT_TYPE_SVIDEO},
+    {"dsVIDEOPORT_TYPE_1394", (int32_t)dsVIDEOPORT_TYPE_1394},
+    {"dsVIDEOPORT_TYPE_DVI", (int32_t)dsVIDEOPORT_TYPE_DVI},
+    {"dsVIDEOPORT_TYPE_COMPONENT", (int32_t)dsVIDEOPORT_TYPE_COMPONENT},
+    {"dsVIDEOPORT_TYPE_HDMI", (int32_t)dsVIDEOPORT_TYPE_HDMI},
+    {"dsVIDEOPORT_TYPE_HDMI_INPUT", (int32_t)dsVIDEOPORT_TYPE_HDMI_INPUT},
+    {"dsVIDEOPORT_TYPE_INTERNAL", (int32_t)dsVIDEOPORT_TYPE_INTERNAL},
+    {"dsVIDEOPORT_TYPE_MAX", (int32_t)dsVIDEOPORT_TYPE_MAX},
+    {NULL, -1}
+};
+
+// Mapping table for dsDisplayEvent_t
+const static stringkey_mapping displayEventMappingTable[] = {
+    {"dsDISPLAY_EVENT_CONNECTED", (int32_t)dsDISPLAY_EVENT_CONNECTED},
+    {"dsDISPLAY_EVENT_DISCONNECTED", (int32_t)dsDISPLAY_EVENT_DISCONNECTED},
+    {"dsDISPLAY_RXSENSE_ON", (int32_t)dsDISPLAY_RXSENSE_ON},
+    {"dsDISPLAY_RXSENSE_OFF", (int32_t)dsDISPLAY_RXSENSE_OFF},
+    {"dsDISPLAY_HDCPPROTOCOL_CHANGE", (int32_t)dsDISPLAY_HDCPPROTOCOL_CHANGE},
+    {"dsDISPLAY_EVENT_MAX", (int32_t)dsDISPLAY_EVENT_MAX},
+    {NULL, -1}
+};
+
+// Mapping table for dsVideoAspectRatio_t
+const static stringkey_mapping dsVideoAspect_RatioMappingTable[] = {
+    {"dsVIDEO_ASPECT_RATIO_4x3", (int32_t)dsVIDEO_ASPECT_RATIO_4x3},
+    {"dsVIDEO_ASPECT_RATIO_16x9", (int32_t)dsVIDEO_ASPECT_RATIO_16x9},
+    {"dsVIDEO_ASPECT_RATIO_MAX", (int32_t)dsVIDEO_ASPECT_RATIO_MAX},
+    {NULL, -1}
+};
+
+static char* mapKeyToString(const stringkey_mapping* keyMappingTable, int32_t keyCode) {
+    if (keyMappingTable == NULL) {
+        return NULL;
+    }
+
+    for (int32_t i = 0; keyMappingTable[i].stringValue != NULL; i++) {
+        if (keyMappingTable[i].keyCode == keyCode) {
+            return keyMappingTable[i].stringValue;
+        }
+    }
+
+    return "Unknown value";
+}
+
+/*callback*/
+void DisplayEventCallback(intptr_t handle, dsDisplayEvent_t event, void* eventData)
+{
+    UT_LOG_INFO("DisplayEventCallback dsDisplayEvent_t:[%s] ",mapKeyToString(displayEventMappingTable, event));
+}
+
+void dsDisplay_Init()
+{
+    dsError_t status   = dsERR_NONE;
+
+    UT_LOG_INFO("Calling dsDisplayInit()");
+    status = dsDisplayInit();
+    UT_LOG_INFO("Result dsDisplayInit() dsError_t=[%s]", mapKeyToString(errorMappingTable, status));
+    UT_ASSERT_EQUAL_FATAL(status, dsERR_NONE);
+
+}
+
+void dsDisplay_Term()
+{
+    dsError_t status   = dsERR_NONE;
+
+    UT_LOG_INFO("Calling dsDisplayTerm()");
+    status = dsDisplayTerm();
+    UT_LOG_INFO("Result dsDisplayTerm() dsError_t=[%s]",  mapKeyToString(errorMappingTable, status));
+    UT_ASSERT_EQUAL_FATAL(status, dsERR_NONE);
+}
+
+static void dsGetDisplay_getHandle()
+{
+    dsError_t status   = dsERR_NONE;
+    int32_t choice,port;
+    intptr_t displayHandle = -1;
+
+    dsDisplay_Init();
```

Comment:
Do we need Init Here ?
We will be calling Init as first step and get Handle as next step right ?

Reply:
Done